src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java

```diff
@@ -73,6 +75,57 @@ public String apply(String updatedDate, int currentIndex, Translation translatio
       ZonedDateTime originDateTime = ZonedDateTime.parse(updatedDate, originFormatter);
       return targetFormatter.format(originDateTime);
     }
+  },
+
+  /**
+   * Forty character positions (00-39) that provide coded information about the record as a whole and about special
+   * bibliographic aspects of the item being cataloged.
+   * These coded data elements are potentially useful for retrieval and data management purposes.
+   * Format:
+   * 00-05 - Metadata.createdDate field in yymmdd format
+   * 06 is set to | (pipe character)
+   * 07-10 to publication[0] dateOfPublication if can be formatted else |||| (four pipe characters)
+   * 11-14 to publication[1] dateOfPublication if can be formatted else |||| (four pipe characters)
+   * 18-22 - each field set to |
+   * 23-29 - each field to be blank
+   * 30-34 - each field set to |
+   * 35-37 - if languages array is empty set it to "und",
+   * if one element, use it to populate the field (it should be 3 letter language code),
+   * if the array contains more than one language, then set it to "mul"
+   * 38-39 - each field set to |
+   */
+  SET_FIXED_LENGTH_DATA_ELEMENTS() {
+    private transient DateTimeFormatter originCreatedDateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+    private transient DateTimeFormatter targetCreatedDateFormatter = DateTimeFormatter.ofPattern("yyMMdd");
+    private String fieldPattern = "%s|%s%s||||||||       |||||%s||";
+
+    @Override
+    public String apply(String originCreatedDate, int currentIndex, Translation translation, ReferenceData referenceData, Metadata metadata) {
+      String createdDateParam = targetCreatedDateFormatter.format(ZonedDateTime.parse(originCreatedDate, originCreatedDateFormatter));
+
+      String publicationDate0Param = "||||";
+      String publicationDate1Param = "||||";
+      if (metadata.getData().containsKey("datesOfPublication")) {
+        List<String> publicationDates = (List<String>) metadata.getData().get("datesOfPublication").getData();
+        if (publicationDates.size() == 1) {
+          publicationDate0Param = publicationDates.get(0);
```

Comment:
if the publication date is not 4 characters long and is just "123". It will not fill the 4 characters right? Then won't the formatting get affected?

Reply:
I just thought the date always takes 4 chars, I will fix it. The question what if the date is more then 4 chars?

Reply:
Magda answered in Jira - the date cannot be formatted (because is too short or too long) then replace it ||||